dask/array/linalg.py

```diff
@@ -328,3 +329,82 @@ def svd(a, name=None):
     dask.array.linalg.tsqr: Actual implementation with citation
     """
     return tsqr(a, name, compute_svd=True)
+
+
+def lu(a):
+    import scipy.linalg
+
+    if a.ndim != 2:
+        raise ValueError
+
+    vdim = len(a.chunks[0])
+    hdim = len(a.chunks[1])
+
+    token = tokenize(a)
+    name_lu = 'lu-lu-' + token
+
+    name_p = 'lu-p-' + token
+    name_l = 'lu-l-' + token
+    name_u = 'lu-u-' + token
+
+    dsk = {}
+
+    for i in range(min(vdim, hdim)):
+
+        if i == 0:
+            target = (a.name, i, i)
+        else:
+            prevs = [(np.dot, (name_l, i, p),
+                              (name_u, p, i)) for p in range(i)]
+            prevs = reduce(lambda x, y: (operator.add, x, y), prevs)
```

Comment:
I think that this should either be `prevs = (sum, prevs)` or it should be many small tasks within the graph.